frontend/src/Crossword.jsx

```diff
@@ -32,6 +48,69 @@ const Crossword = (props) => {
     setCanvas(canvas);
   }, []);
 
+  const cursorKey = useCallback(
+    (key) => {
+      let [r, c] = cursorRC;
+
+      if (key === "ArrowDown" || key === "Down") {
+        r += 1;
+      } else if (key === "ArrowUp" || key === "Up") {
+        r -= 1;
+      } else if (key === "ArrowLeft" || key === "Left") {
+        c -= 1;
+      } else if (key === "ArrowRight" || key === "Right") {
+        c += 1;
+      } else {
+        return false;
+      }
+
+      if (r >= 0 && r < squares.length && c >= 0 && c < squares[r].length)
+        setCursorRC([r, c]);
+      return true;
+    },
+    [cursorRC]
+  );
+
+  const escSwitchMode = (key) => {
+    if (key === "Escape" || key === "Esc") {
+      setMode(DRAW);
+      return true;
+    }
+
+    return false;
+  };
+
+  const letterKey = useCallback(
+    (key) => {
+      const updateAndSend = (newValue) => {
+        const [r, c] = cursorRC;
+        const newLetters = [...letters];
+        newLetters[coordGrid[r][c]].l = newValue;
+        setLetters(newLetters);
+
+        sendEvent({
+          action: "WRITE_EVENT",
+          event: { l: newValue, r, c }
+        });
+      };
+
+      const isLetter =
+        key.length === 1 && key.toUpperCase().match(/[A-Z|Å|Ä|Ö]/i);
+      if (isLetter) {
+        updateAndSend(key.toUpperCase());
+
+        return true;
+      } else if (key === "Backspace") {
+        updateAndSend("");
```

Comment:
I think we need to prevent default as well somewhere. In Firefox, hitting backspace takes me back to the previous page (applicable for browser where this backspace behavior is still the case).

Reply:
Doesn't happen for me in ff 74.0. Already have one preventdefault here https://github.com/thesauri/korsord/pull/20/files/71300451dfdcb87e964fd66a35a850ff0d73a3c3#diff-1ec7bbb2d4f5a1c6526cbf95777c02d6R277

I'll add one for backspace :+1: